Automattic/wp-calypso
# Replace manual monthly price calculations with a function calculateMonthlyPrice() that takes different terms into consideration

Most functions concerned with calculating monthly price or discount use the following hardcoded formula: `parseFloat( ( plan.rawDiscount / 12 ).toFixed( 2 ) )`. It returns invalid results for monthly plans, and also for the new 2-year plans. This PR replaces that with a function that takes plan term into consideration.

Test plan:
1. Run unit tests.
2. Go to `http://calypso.localhost:3000/plans` for both WP and Jetpack sites and confirm that prices are intact
3. Buy just a domain without a site, and confirm that `DomainToPlanNudge` nudge displays the correct price
4. Go to post share section as shown below both on free plan on both WP and Jetpack site and confirm that `UpgradeToPremiumNudge` displays the correct price

![screen shot on 2018-03-23 at 13 28 39](https://user-images.githubusercontent.com/205419/37829137-296579b4-2e9e-11e8-9d81-426d08117f9d.png)
